### DISCUSSION TITLE
woodgern confusables package

### DISCUSSION THREAD

**Discussion by rachmadaniHaryono** [ThioJoe/YT-Spammer-Purge#666] [Q&A] [answered]
i just notice that the repo contain confusableCustom subdirectory which i believe is modified from this https://github.com/woodgern/confusables

is it actually ok to do that?

i assume you have to include license file from confusables because this part of mit license

> The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

https://github.com/woodgern/confusables/blob/f1d9302486f0ad724444c44732039853edc07424/LICENSE#L12-L13

if you need confusables with custom data you can fork the repo and use forked repo github url to requirements like this

`git+https://github.com/woodgern/confusables.git`

disclaimer: i am not a lawyer and this is not legal advice

https://en.wikipedia.org/wiki/IANAL

**Comment by ThioJoe** [comment, marked as answer]
Hm hadn't even thought to fork it, but that probably does make more sense

**Comment by ThioJoe** [comment]
I went and added the license along with the files in https://github.com/ThioJoe/YT-Spammer-Purge/commit/45d7003c2daec2b514260ca499b67f11c30dedbc

Apparently there's a thing called "Git Submodules" that lets you include other repos in your own somehow, so I'll have to look into that. I did go and fork it here: https://github.com/ThioJoe/confusables

**Reply by Firecul** [reply]
I was going to make a pr to use submodules for the spam lists for the other repos but didn't want to get into changing the code to accommodate that, I'll have to take another look at it if it's something you'd consider?

**Reply by ThioJoe** [reply]
Hm yea it probably would make sense, at least so the included default spam lists are up to date every time, since I kind of keep forgetting to update them (Even though it doesn't really matter and the program checks the spam lists repo on first start. The default lists are basically just in there in case somehow that fails).

If it would be really involved though I'd probably better learn it myself, so it depends.

**Reply by Firecul** [reply]
Submodules are pretty easy. I have a question though so I'm going to start another discussion on it.
